src/eos-updater-fetch.c

```diff
@@ -820,6 +849,34 @@ unschedule_download (FetchData     *fetch_data,
   return TRUE;
 }
 
+static void
+cancel_if_download_forbidden_cb (GObject    *obj,
+                                 GParamSpec *pspec,
+                                 gpointer    user_data)
+{
+  MwscScheduleEntry *entry = MWSC_SCHEDULE_ENTRY (obj);
+  GCancellable *cancellable = user_data;
+
+  if (!mwsc_schedule_entry_get_download_now (entry))
+    {
+      g_message ("Fetch cancelled because Mogwai's scheduled entry %s can no "
+                 "longer download.", mwsc_schedule_entry_get_id (entry));
```

Comment:
This function needs a FIXME comment, since Mogwai can actually toggle the status of `download-now` several times during the lifetime of a schedule entry. For example, if an OS update was scheduled and downloading, and then a higher-priority update was scheduled by flatpak, Mogwai would pause the OS update (by setting its `download-now` to false), do the flatpak update, and then could resume the OS update (setting `download-now` to true again) once the flatpak update’s schedule entry was removed.

Supporting resuming updates in eos-updater is a fairly big change, though, and not one I’d expect for this PR. Cancelling the update and re-polling next time the updater is triggered should be OK for now; since OSTree downloads files individually, this should allow progress to be saved between update attempts, if none of them are allowed to complete.

Anyway, in the meantime, a comment here like the following would be good:
```
/* FIXME: Eventually add support for resuming OS update downloads when allowed
 * to by the scheduler. When that happens `get_download_now()` will return
 * %TRUE again; while waiting for that to happen, eos-updater should stay in the
 * `fetching` state. Currently, we go to the `error` state instead. */
```